diff --git a/ssl/encrypted_client_hello.cc b/ssl/encrypted_client_hello.cc
index a5492e9..ebf5f85 100644
--- a/ssl/encrypted_client_hello.cc
+++ b/ssl/encrypted_client_hello.cc
@@ -746,7 +746,7 @@ static bool setup_ech_grease(SSL_HANDSHAKE *hs) {
   // estimate maximum_name_length to be between 32 and 100 bytes. Then round up
   // to a multiple of 32, to match draft-ietf-tls-esni-13, section 6.1.3.
   const size_t payload_len =
-      32 * random_size(128 / 32, 224 / 32) + aead_overhead(aead);
+      32 * random_size(192 / 32, 224 / 32) + aead_overhead(aead);
   bssl::ScopedCBB cbb;
   CBB enc_cbb, payload_cbb;
   uint8_t *payload;
